README.md

```diff
@@ -1,123 +1,323 @@
 gcloud config set project <project_id>
-sh deploy-data-tokeninzation-solution.sh
+sh deploy-data-tokeninzation-solution-v2.sh
 ```
 
- You can run some quick [validations](https://cloud.google.com/solutions/validating-de-identified-data-bigquery-re-identifying-pii-data#validating_the_de-identified_dataset_in_bigquery) in BigQuery table to check on tokenized data.  
+Script (deploy-data-tokenization-solution-v2.sh) handles following topics:
+
+* Create a service account for running the DLP pipeline (creates a custom role).
+
+* Emit a set_env.sh that can be used to set temporary environment variables while triggering the DLP pipelines.
+
+* Create a bucket ({project-id}-demo-data) in us-central1 and [uploads a sample dataset](http://storage.googleapis.com/dataflow-dlp-solution-sample-data/sample_data_scripts.tar.gz) with <b>mock</b> PII data.
+
+* Create a BigQuery dataset in US (demo_dataset) to store the tokenized data.
+
+* Create a [KMS wrapped key(KEK)](https://cloud.google.com/kms/docs/envelope-encryption) by creating an automatic [TEK](https://cloud.google.com/solutions/de-identification-re-identification-pii-using-cloud-dlp#token_encryption_keys) (Token Encryption Key).
 
-For re-identification (getting back the original data in a Pub/Sub topic), please follow this instruction [here](https://cloud.google.com/solutions/validating-de-identified-data-bigquery-re-identifying-pii-data#re-identifying_the_dataset_from_bigquery).  
+* Create DLP [inspect, de-identification and re-identification templates](https://cloud.google.com/solutions/creating-cloud-dlp-de-identification-transformation-templates-pii-dataset#creating_the_cloud_dlp_templates) with the KEK and crypto based transformations identified in this [section of the guide](https://cloud.google.com/solutions/de-identification-re-identification-pii-using-cloud-dlp#determining_transformation_type)
+
+4. Run set_env.sh
+```
+source set_env.sh
+```
 
-## V2 Solution By Using In Built Java Beam Transform
-This part of the repo provides a reference implementation to process large scale files for  any DLP transformation like Inspect, Deidentify or ReIdentify.  Solution can be used for CSV / Avro / [JSONL](https://jsonlines.org) files stored in either GCS or AWS S3 bucket. It uses State and Timer API for efficient batching to process the files in optimal manner. 
 
-## Build and Run
+### Compile the code
+
 ```
 gradle spotlessApply
 
 gradle build
+```
+### Run the samples
+#### Inspection
 
-gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs=" --region=<region> --project=<projct_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=gs://<path>.csv --dataset=<name>   --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --DLPMethod=DEID"
 ```
-## S3 Scanner
+gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 \
```

Comment:
highlight that we need to mention region ourselves as between all other parameters it's easy to get lost. Maybe make it bold to attract attention

Reply:
Changed region to environmental variable.